Src/FluentAssertions/Types/CustomAttributeProviderAssertions.cs

```diff
@@ -0,0 +1,151 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using FluentAssertions.Common;
+using FluentAssertions.Execution;
+using FluentAssertions.Primitives;
+
+namespace FluentAssertions.Types;
+
+/// <summary>
+/// Contains a number of methods to assert that a <see cref="ICustomAttributeProvider"/> is in the expected state.
+/// </summary>
+[DebuggerNonUserCode]
+public abstract class CustomAttributeProviderAssertions<TSubject, TAssertions> : ReferenceTypeAssertions<TSubject, TAssertions>
```

Comment:
🤔 Given that this type may contain other assertions that are reusable across the reflection assertions, I think we should name this something like `ReflectionAssertions`
🌱 I hope using inheritance isn't going to cause issues later on.

Reply:
Currently our inheritance hierarchy of type-related assertions follows the inheritance hierarchy of the corresponding BCL types.
I notice that e.g. `Assembly` also implements `ICustomAttributeProvider`.

How difficult would it be to implement this feature without changing our inheritance hierarchy?

It's not show stopper for me, it just feels a little odd to me to mix these concepts.